types/datum.go

```diff
@@ -1415,8 +1415,6 @@ func (d *Datum) ConvertToMysqlYear(sc *stmtctx.StatementContext, target *FieldTy
 		}
 	case KindMysqlTime:
 		y = int64(d.GetMysqlTime().Year())
-	case KindMysqlDuration:
-		y = int64(time.Now().Year())
```

Comment:
Remove this because of https://github.com/pingcap/tidb/issues/25331#issuecomment-896530427.